contrib/labstack/echo.v4/echotrace_test.go

```diff
@@ -448,3 +448,85 @@ func TestAppSec(t *testing.T) {
 		require.True(t, strings.Contains(event.(string), "crs-933-130"))
 	})
 }
+
+func TestIgnoreError(t *testing.T) {
+	tests := []struct {
+		name    string
+		err     error
+		wantErr error
+	}{
+		{
+			name: "echo 404 error",
+			err: &echo.HTTPError{
+				Code:     http.StatusNotFound,
+				Message:  "not found",
+				Internal: errors.New("not found"),
+			},
+			wantErr: nil,
+		},
+		{
+			name: "echo 500 error",
+			err: &echo.HTTPError{
+				Code:     http.StatusInternalServerError,
+				Message:  "internal error",
+				Internal: errors.New("internal error"),
+			},
+			wantErr: &echo.HTTPError{
+				Code:     http.StatusInternalServerError,
+				Message:  "internal error",
+				Internal: errors.New("internal error"),
+			},
+		},
+		{
+			name:    "any error",
+			err:     errors.New("any error"),
+			wantErr: errors.New("any error"),
+		},
+		{
+			name:    "no error",
+			err:     nil,
+			wantErr: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mt := mocktracer.Start()
+			defer mt.Stop()
+
+			router := echo.New()
+			ignoreClientErrors := func(err error) bool {
+				var he *echo.HTTPError
+				if errors.As(err, &he) {
+					if he.Code < 500 && he.Code >= 400 {
+						return true
+					}
+				}
+				return false
+			}
```

Comment:
I think it would probably make more sense if this is provided in each test case input (adding a new field `errorCheck` in the test case struct)